### PR TITLE
feat: add eventgrid subscription to az functions

### LINKED ISSUE #197: Support for endpoint validation in Azure Functions
### Discussed in https://github.com/arcus-azure/arcus.eventgrid/discussions/173

<div type='discussions-op-text'>

<sup>Originally posted by **tomkerkhove** July  9, 2021</sup>
Support for endpoint validation in Azure Functions is something you still have to build yourself and did it a few times recently.

It requires the following:
https://github.com/tomkerkhove/autoscaling-over-time-dashboard/blob/main/src/AzureAutoscalingToolbox.Samples.StatefulAppInstances/Functions/AzureMonitorScaledFunction.cs#L29-L32

Not sure if this should be here or a template though, maybe both?</div>

### PR DESCRIPTION
**Pull request by stijnmoreels**
Extracted the Azure EventGrid subscription validation to a seperate `.Core` project and created a dedicated `.AzureFunctions` project where we provide an extension on the `IFunctionsHostBuilder` so the Azure EventGrid subscription validation and CloudEvents handshake can be used in an Azure Functions context.

Closes #197

### CODE REVIEW

File: build/variables/build.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 variables:
-  DotNet.Sdk.Version: '3.1.201'
+  DotNet.Sdk.Version: '6.0.100'
```
Comment (fgheysels) [inline build/variables/build.yml:2]: Are we doing this together with the implementation of the eventgrid subscription ?
Should'nt we do this in a separate PR ?

File: docs/preview/02-Features/endpoint-validation.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ We provide support for endpoint validation, when implementing your own custom we
 The features described here require the following package:
 
 ```shell
-PM> Install-Package Arcus.EventGrid.WebApi.Security 
+PM> Install-Package Arcus.EventGrid.Security.WebApi
```
Comment (fgheysels) [inline docs/preview/02-Features/endpoint-validation.md:18]: Is this correct ?

Comment (stijnmoreels) [reply docs/preview/02-Features/endpoint-validation.md:18]: Yeah, I changed it so we can use the security namespace for web API and Azure Functions, as in:
* `.Security.WebApi`
* `.Security.AzureFunctions`
This is similar to other Arcus projects, as well, seeing how messaging, security, web API... repo's are structured.

File: build/variables/build.yml (hunk 1 above)
Comment (stijnmoreels) [reply build/variables/build.yml:2]: Yes, I wanted to split it too, but this PR also adds two new projects, and it would need to still use the 'old' .NET Core for that. Seemed a bit an extra, unnecessary step.

File: docs/preview/02-Features/endpoint-validation.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/preview/02-Features/endpoint-validation.md:18]: We haven't yet released this, so we are free to change it.

File: build/variables/build.yml (hunk 1 above)
Comment (stijnmoreels) [reply build/variables/build.yml:2]: But, we can always take a step back, and only focus on the features here. 
It makes me wonder, though, if the migration towards .NET 6 should be considered a breaking change across all repo's. and/or if we should double support both .NET Core and .NET 6....

Comment (stijnmoreels) [reply build/variables/build.yml]: Voila, removed this updates from this PR so it only contains the feature update. That's better indeed. Let's do this .NET 6 update separately. 👍

**Review by fgheysels — APPROVED** [review]
